fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -288,10 +288,12 @@ private void loadProtoExceptRecords(@Nonnull RecordMetaDataProto.MetaData metaDa
 
     private void loadFromProto(@Nonnull RecordMetaDataProto.MetaData metaDataProto,
                                @Nonnull Descriptors.FileDescriptor[] dependencies,
+                               @Nullable Descriptors.FileDescriptor evolvedFileDescriptor,
```

Comment:
I kind of feel like "evolved" doesn't quite capture what this is about. If I am not mistaken, the 99% use case is that this `FileDescriptor`'s `toProto` is byte-for-byte identical with the one in `metaDataProto`. In particular, for just fixing the original issue.

Reply:
I was under impression that this new file descriptor could potentially contain more (evolved ?) record types. Do you have a better term in mind?

Reply:
Oh, yes, indeed. *could*. But often it does not. The reason it's there isn't just so you can do some kind of schema evolution. It's also so you can successfully rendezvous with the generated Java classes in your JVM.

Reply:
More neutral words would be "replacement" or "substitute," but I'm not super happy with them, either.